toolsrc/src/vcpkg/commands.ci.cpp

```diff
@@ -436,36 +461,16 @@ namespace vcpkg::Commands::CI
                 serialize_options.randomizer = &randomizer_instance;
             }
 
-            auto action_plan = [&]() {
-                int iterations = 0;
-                do
-                {
-                    bool inconsistent = false;
-                    auto action_plan = pgraph.serialize(serialize_options);
-
-                    for (auto&& action : action_plan)
-                    {
-                        if (auto p = action.install_action.get())
-                        {
-                            p->build_options = install_plan_options;
-                            if (Util::Sets::contains(exclusions_set, p->spec.name()))
-                            {
-                                p->plan_type = InstallPlanType::EXCLUDED;
-                            }
-
-                            for (auto&& feature : split_specs->features[p->spec])
-                                if (p->feature_list.find(feature) == p->feature_list.end())
-                                {
-                                    pgraph.install({p->spec, feature});
-                                    inconsistent = true;
-                                }
-                        }
-                    }
+            auto action_plan = Dependencies::PackageGraph::create_feature_install_plan(
+                provider, var_provider, split_specs->unknown, {}, {});
 
-                    if (!inconsistent) return action_plan;
-                    Checks::check_exit(VCPKG_LINE_INFO, ++iterations < 100);
-                } while (true);
-            }();
+            for (auto&& action : action_plan)
+            {
+                if (auto action_ptr = action.install_action.get())
+                {
+                    action_ptr->build_options = install_plan_options;
```

Comment:
This should check and mark exclusions (similar to previous code).

This handles the case of `A -> B -> C` with `B` marked excluded and `A` having unknown results.